blogposts/2021/how-we-built-batch-changes.md

```diff
@@ -0,0 +1,197 @@
+allowed us to go faster and get a _valuable_ tool into the hands of our
+customers even earlier.
+
+So far, so good, right? Yes, except, you know, [naming is
+hard](https://martinfowler.com/bliki/TwoHardThings.html) and while we could've
+made the old name for the feature — _Automation_ — work we decided to rename it.
+Take a look at [PR #3][p3] for the first of many in February 2020 that changed
+"[Aa]utomation" to "[Cc]ampaigns" in our codebase.
+
+What followed were months of fine-tuning, user testing, shipping, tweaking -
+_iterating_. We added more features, fixed a lot of bugs, made things go faster.
+
+But — and by now you should know where this is going — something felt _off_.
+While Campaigns worked and we had customers using it and saying they loved it,
+we also noticed that our colleagues had problems using it, often hesitating to
+admit that they were confused with the workflow. Some of the questions we got:
+why JSON (yes, we made users write JSON by hand)? Why can't I put everything in
+that JSON file, why do I have to switch to the web UI? And what if I want to
+update the changes?
+
+I wouldn't say we were stuck again - we were constantly shipping new features
+and improvements. But it also started dawning on us that maybe we've exhausted
+one idea. Tweaking wouldn't get us out of this, nor a sledgehammer. But maybe if
+we look at it from a different angle? What if instead of requiring users say
+what should happen to produce a campaign, we switch to a _declarative_ model and
+let them describe what the campaign should like in a YAML file?
+
+[This pull request](https://github.com/sourcegraph/sourcegraph/pull/10921) shows
+how we asked "what if?": we wrote the documentation first (yes, README-driven
+development) and showed it to teammates and colleagues. The time it took a
+colleague to go from "ok, tell me what campaigns are" to "ohhh, I get it! nice"
+was reduced tenfold. Very much to our relief, since the changes necessary to
+make the documentation a reality were huge. We had to build a distributed,
+_declarative_ system that manages hundreds and thousands of pull and merge
+requests across different code hosts.
+
+Once the idea was validated by a lot "this totally makes sense" from potential
+users we got to work. Look at [this PR][pr4b], number 4b in this series if you
+will, to see an example of what "declarative" means and how we reconcile the
+current and the desired state (as described by a user) of a changeset. (Sidenote
+for the curious: the code today is [even cooler][evencoolercode], take a look).
+
+That brings us to the last of the five pull requests.
+
+Towards the end of last year we ripped the "beta" label off of Campaigns and
+started to concentrate on getting more customers to use it: writing better (or
+even any) documentation, improving the onboarding process, providing
+troubleshooting help, fixing bugs and edge cases.
+
+At the same time our fast-growing product marketing team started working towards
```

Comment:
```suggestion
At the same time, our fast-growing product marketing team started working towards
```
This comma means there's only one way to parse this. Otherwise I have to read ahead to know how to parse it. Because maybe you were saying, "At the same time *that* our fast-growing..." in which case the sentence ends quite abruptly because the second half of it is missing.